vertx-starter-generator/src/main/java/io/vertx/starter/generator/service/ArchiveService.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2017 The original author or authors
+ * ------------------------------------------------------
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Apache License v2.0 which accompanies this distribution.
+ *
+ *     The Eclipse Public License is available at
+ *     http://www.eclipse.org/legal/epl-v10.html
+ *
+ *     The Apache License v2.0 is available at
+ *     http://www.opensource.org/licenses/apache2.0.php
+ *
+ * You may elect to redistribute this code under either of these licenses.
+ */
+package io.vertx.starter.generator.service;
+
+import io.vertx.core.Vertx;
+import io.vertx.core.eventbus.Message;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import org.zeroturnaround.zip.ZipUtil;
+
+import java.io.File;
+
+public class ArchiveService {
+
+    private final Logger log = LoggerFactory.getLogger(ArchiveService.class);
+
+    private Vertx vertx;
+
+    public ArchiveService(Vertx vertx) {
+        this.vertx = vertx;
+    }
+
+    public void archive(Message<JsonObject> message) {
+        JsonObject metadata = message.body();
+        String baseDir = metadata.getString("baseDir");
+        String rootDir = metadata.getString("rootDir");
```

Comment:
shouldn't we use `tmpdir` here? if you're deployed on a system where the user has no permissions to write to disk i think this can cause some problems